src/storage/src/sst.rs

```diff
@@ -202,21 +221,58 @@ impl FileHandle {
 struct FileHandleInner {
     meta: FileMeta,
     compacting: AtomicBool,
+    deleted: AtomicBool,
+    sst_layer: AccessLayerRef,
+    file_purger: FilePurgerRef,
+}
+
+impl Drop for FileHandleInner {
+    fn drop(&mut self) {
+        if self.deleted.load(Ordering::Relaxed) {
```

Comment:
Do we need to check `compacting` too? Is it possible that a compacting file is marked to be deleted?

Reply:
Currently no, files are only marked as deleted iff compaction succeeds.

Reply:
> Do we need to check `compacting` too? Is it possible that a compacting file is marked to be deleted?

Currently no, files are only marked as deleted iff compaction succeeds.